### DISCUSSION TITLE
Support AWS IAM-based auth for elasticache redis 7.0

### DISCUSSION THREAD

**Discussion by servionsolutions** [redisson/redisson#4710] [Ideas]
Hello!  It would be great if redisson could support the new IAM-based authentication tokens for AWS elasticache.  This is a new feature that AWS has released in their redis engine v7:
[IAM release announcement](https://aws.amazon.com/about-aws/whats-new/2022/11/amazon-elasticache-supports-iam-authentication-redis-clusters/)
[IAM docs for elasticache](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/auth-iam.html)

It looks like it may require the ability to rotate out short lived tokens, thus we may need a way to provide a credential provider to redisson.  Also would be good to make sure that long/persistent connections are able to continue to be supported beyond 12/24 hour windows.

**Comment by mrniko** [comment]
It's a good idea! Can you create a feature request issue?

**Reply by servionsolutions** [reply]
Yes :) https://github.com/redisson/redisson/issues/4713

**Reply by mrniko** [reply]
Thank you!
